libs/ledger-live-common/src/apps/logic.ts

```diff
@@ -437,12 +437,13 @@ export function getBlockSize(state: State): number {
 export const isIncompleteState = (state: State): boolean => state.installed.some(a => !a.name);
 // calculate if a given state (typically a predicted one) is out of memory (meaning impossible to reach with a device)
 export const isOutOfMemoryState = (state: State): boolean => {
+  const { customImageBlocks } = state;
   const blockSize = getBlockSize(state);
   const totalBytes = state.deviceModel.memorySize;
   const totalBlocks = Math.floor(totalBytes / blockSize);
   const osBytes = (state.firmware && state.firmware.bytes) || 0;
   const osBlocks = Math.ceil(osBytes / blockSize);
-  const appsSpaceBlocks = totalBlocks - osBlocks;
+  const appsSpaceBlocks = totalBlocks - osBlocks - customImageBlocks;
```

Comment:
shouldn't this `isOutOfMemoryState` function rely on `distribute` to ensure that all these calculations are only implement in one place and avoid the apparition of such discrepancies 
(in this case the computation of `appsSpaceBlocks` was implemented in two places and at some point there was a divergence) ?

Reply:
Good point here, I'll do it 👍